diff --git a/CHANGELOG b/CHANGELOG
index ca290f9..5e3a9ec 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -38,6 +38,7 @@
 - add dynamic logging (adapted from v4 patch from Jeff Moyer).
 - fix recursive loopback mounts (Matthias Koenig).
 - add map re-load to verbose logging.
+- fix handling of LDAP base dns with spaces.
 
 18/06/2007 autofs-5.0.2
 -----------------------
diff --git a/lib/master_tok.l b/lib/master_tok.l
index 00cd223..48bc233 100644
--- a/lib/master_tok.l
+++ b/lib/master_tok.l
@@ -109,6 +109,7 @@ AT_O		([oO])
 AT_C		([cC])
 DNATTRSTR	{AT_CN}|{AT_NMN}|{AT_AMN}|{AT_OU}|{AT_DC}|{AT_O}|{AT_C}
 DNNAMESTR	([[:alnum:]_.\-]+)
+DNNAMETRM	(,|{WS}{NL}|{WS}#.*|\x00)
 
 INTMAP		(-hosts|-null)
 MULTI		((multi)(,(sun|hesiod))?(:{OPTWS}|{WS}))
@@ -281,7 +282,7 @@ OPTTOUT		(-t{OPTWS}|-t{OPTWS}={OPTWS}|--timeout{OPTWS}|--timeout{OPTWS}={OPTWS})
 		return EQUAL;
 	}
 
-	{DNNAMESTR} {
+	{DNNAMESTR}/{DNNAMETRM} {
 		strcpy(master_lval.strtype, master_text);
 		return DNNAME;
 	}
