diff --git a/CHANGELOG b/CHANGELOG
index 0e9dc51..054d4df 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -21,6 +21,7 @@
 - if there's no "automount" entry in nsswitch.conf use "files" source.
 - add LDAP schema discovery if no schema is configured.
 - add random selection as a master map entry option.
+- fix couple of edge case parse fails of timeout option.
 
 18/06/2007 autofs-5.0.2
 -----------------------
diff --git a/lib/master_tok.l b/lib/master_tok.l
index 013a15a..2735223 100644
--- a/lib/master_tok.l
+++ b/lib/master_tok.l
@@ -313,7 +313,7 @@ OPTTOUT		(-t{OPTWS}|-t{OPTWS}={OPTWS}|--timeout{OPTWS}|--timeout{OPTWS}={OPTWS})
 		return(DDASH);
 	}
 
-	{OPTTOUT} { return(OPT_TIMEOUT); }
+	{OPTTOUT}/{NUMBER} { return(OPT_TIMEOUT); }
 
 	{NUMBER} {
 		master_lval.longtype = atol(master_text);
