--- autofs-5.1.1/samples/autofs.service.in.orig	2016-03-29 22:03:38.660685363 +0200
+++ autofs-5.1.1/samples/autofs.service.in	2016-03-30 15:53:17.951325347 +0200
@@ -1,6 +1,6 @@
 [Unit]
 Description=Automounts filesystems on demand
-After=network.target ypbind.service sssd.service network-online.target
+After=network.target ypbind.service sssd.service network-online.target nslcd.service
 Wants=network-online.target
 
 [Service]
