diff --git a/CHANGELOG b/CHANGELOG
index 172d0cd..6931791 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -1,3 +1,7 @@
+??/??/2007 autofs-5.0.3
+-----------------------
+- include krb5.h in lookup_ldap.h (some openssl doesn't implicitly include it).
+
 18/06/2007 autofs-5.0.2
 -----------------------
 - fix return check for getpwuid_r and getgrgid_r.
diff --git a/include/lookup_ldap.h b/include/lookup_ldap.h
index e1c5b4e..0a9deca 100644
--- a/include/lookup_ldap.h
+++ b/include/lookup_ldap.h
@@ -7,6 +7,7 @@
 #include <openssl/err.h>
 #include <sasl/sasl.h>
 #include <libxml/tree.h>
+#include <krb5.h>
 #endif
 
 struct lookup_context {
