diff --git a/CHANGELOG b/CHANGELOG
index 0340940..ddfa6f1 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -29,6 +29,7 @@
 - re-read config on HUP signal.
 - add LDAP_URI, LDAP_TIMEOUT and LDAP_NETWORK_TIMEOUT configuration options.
 - fix forground logging and add option to man page.
+- remove unjustified, nasty comment about krb5 package.
 
 18/06/2007 autofs-5.0.2
 -----------------------
diff --git a/daemon/automount.c b/daemon/automount.c
index 51f6a8b..930b13f 100644
--- a/daemon/automount.c
+++ b/daemon/automount.c
@@ -1263,22 +1263,6 @@ void *handle_mounts(void *arg)
 	}
 
 	pthread_cleanup_pop(1);
-
-	/*
-	 * A cowboy .. me!
-	 * That noise yu ear aint spuurs sonny!!
-	 *
-	 * The libkrb5support destructor called indirectly through
-	 * libgssapi_krb5 which is used bt libkrb5 (somehow) must run
-	 * to completion before the last thread using it exits so
-	 * that it's per thread data keys are deleted or we get a
-	 * little segfault at exit. So much for dlclose being
-	 * syncronous.
-	 *
-	 * So, the solution is a recipe for disaster.
-	 * Hope we don't get a really busy system!
-	 */
-	/*sleep(1);*/
 	sched_yield();
 
 	return NULL;
