--- autofs-5.1.4/samples/autofs.service.in.orig	2018-04-11 05:33:10.457186708 +0200
+++ autofs-5.1.4/samples/autofs.service.in	2018-04-11 17:15:47.396705425 +0200
@@ -1,6 +1,6 @@
 [Unit]
 Description=Automounts filesystems on demand
-After=network.target ypbind.service sssd.service network-online.target remote-fs.target
+After=network.target ypbind.service sssd.service network-online.target remote-fs.target nslcd.service
 Wants=network-online.target rpc-statd.service rpcbind.service
 
 [Service]
