--- autofs-5.1.5/samples/autofs.service.in.orig	2018-11-24 22:55:08.414435285 +0100
+++ autofs-5.1.5/samples/autofs.service.in	2018-11-25 08:34:53.977371427 +0100
@@ -1,6 +1,6 @@
 [Unit]
 Description=Automounts filesystems on demand
-After=network.target ypbind.service sssd.service network-online.target remote-fs.target rpc-statd.service rpcbind.service
+After=network.target ypbind.service sssd.service network-online.target remote-fs.target rpc-statd.service rpcbind.service nslcd.service
 Wants=network-online.target rpc-statd.service rpcbind.service
 
 [Service]
