--- autofs-3.1.4/modules/Makefile	Fri Jan 21 18:21:32 2000
+++ autofs-3.1.4/modules/Makefile	Fri Jan 21 18:21:45 2000
@@ -43,7 +43,7 @@
 # mount_smbfs.so is an obsolete module which must be removed
 install: all
 	install -d -m 755 $(INSTALLROOT)$(autofslibdir)
-	install -c $(MODS) -m 644 $(INSTALLROOT)$(autofslibdir)
+	install -c $(MODS) -m 755 $(INSTALLROOT)$(autofslibdir)
 	-rm -f $(INSTALLROOT)$(autofslibdir)/mount_smbfs.so
 
 #
