--- ./daemon/mount.c.orig	Sat Jun 23 13:55:53 2001
+++ ./daemon/mount.c	Sat Jun 23 13:56:22 2001
@@ -21,6 +21,7 @@
 
 #include <syslog.h>
 #include <stdlib.h>
+#include <string.h>
 #include "automount.h"
 
 /* These filesystems are known not to work with the "generic" module */
--- ./modules/parse_sun.c.orig	Sat Jun 23 13:57:04 2001
+++ ./modules/parse_sun.c	Sat Jun 23 14:02:22 2001
@@ -17,6 +17,7 @@
 
 #include <stdio.h>
 #include <malloc.h>
+#include <alloca.h>
 #include <errno.h>
 #include <netdb.h>
 #include <fcntl.h>
--- ./modules/mount_nfs.c.orig	Sat Jun 23 14:02:40 2001
+++ ./modules/mount_nfs.c	Sat Jun 23 14:04:29 2001
@@ -17,6 +17,7 @@
 
 #include <stdio.h>
 #include <malloc.h>
+#include <alloca.h>
 #include <errno.h>
 #include <netdb.h>
 #include <fcntl.h>
--- ./modules/mount_generic.c.orig	Sat Jun 23 14:04:41 2001
+++ ./modules/mount_generic.c	Sat Jun 23 14:04:59 2001
@@ -16,6 +16,7 @@
 
 #include <stdio.h>
 #include <malloc.h>
+#include <alloca.h>
 #include <errno.h>
 #include <fcntl.h>
 #include <unistd.h>
--- ./modules/mount_ext2.c.orig	Sat Jun 23 14:05:18 2001
+++ ./modules/mount_ext2.c	Sat Jun 23 14:05:35 2001
@@ -16,6 +16,7 @@
 
 #include <stdio.h>
 #include <malloc.h>
+#include <alloca.h>
 #include <errno.h>
 #include <fcntl.h>
 #include <unistd.h>
--- ./Makefile.rules.orig	Sat Jun 23 14:07:08 2001
+++ ./Makefile.rules	Sat Jun 23 14:08:24 2001
@@ -15,13 +15,13 @@
 # The STRIP defined here *must not* remove any dynamic-loading symbols
 
 ifdef DEBUG
-CFLAGS    = -O2 -g -DDEBUG
-LDFLAGS   = -g
+CFLAGS    = -O3 -fomit-frame-pointer -Wall
+LDFLAGS   = -ggdb
 STRIP     = :
 else
-CFLAGS    = -O3 -fomit-frame-pointer -Wall
-LDFLAGS   = -s
-STRIP     = strip --strip-debug
+CFLAGS    = $(RPM_OPT_FLAGS)
+LDFLAGS   = 
+STRIP     = :
 endif
 
 CC        = gcc
