diff --git a/CHANGELOG b/CHANGELOG
index 20562bd..da8c599 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -14,6 +14,7 @@
 - fix version passed to get_supported_ver_and_cost.
 - mark map instances stale so they aren't "cleaned" during updates.
 - fix large file compile time option.
+- don't fail on empty master map.
 
 18/06/2007 autofs-5.0.2
 -----------------------
diff --git a/lib/master.c b/lib/master.c
index 4d31959..9f24f7e 100644
--- a/lib/master.c
+++ b/lib/master.c
@@ -803,7 +803,7 @@ int master_read_master(struct master *master, time_t age, int readall)
 	if (list_empty(&master->mounts)) {
 		master_mutex_unlock();
 		error(LOGOPT_ANY, "no mounts in table");
-		return 0;
+		return 1;
 	}
 
 	master_mutex_unlock();
