--- autofs-4.0.0pre7/modules/parse_sun.c	Tue Aug  1 17:42:17 2000
+++ autofs-4.0.0pre7-junk/modules/parse_sun.c	Tue Aug  1 17:40:50 2000
@@ -466,6 +466,7 @@
     int len = strlen(options)+1;
 
     noptions = np = alloca(len);
+    memset(noptions, '\0', len);
 
     /* Extract fstype= pseudo option */
     for(comma = options; *comma != '\0'; ) {
@@ -489,7 +490,7 @@
 	np += comma-cp+1;
       }
     }
-    np[-1] = '\0';
+    if(np > noptions) np[-1] = '\0';
 
     options = noptions;
   }
