diff --git a/CHANGELOG b/CHANGELOG
index fdd07d1..fe7ae00 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -33,6 +33,7 @@
 - fix deadlock in submount mount module.
 - fix lack of ferror() checking when reading files.
 - fix typo in autofs(5) man page.
+- fix map entry expansion when undefined macro is present.
 
 18/06/2007 autofs-5.0.2
 -----------------------
diff --git a/modules/parse_sun.c b/modules/parse_sun.c
index 5e14c75..079bda6 100644
--- a/modules/parse_sun.c
+++ b/modules/parse_sun.c
@@ -186,8 +186,7 @@ int expandsunent(const char *src, char *dst, const char *key,
 						dst += l;
 					}
 					len += l;
-				} else
-					return 0;
+				}
 				src = p + 1;
 			} else {
 				p = src;
@@ -201,8 +200,7 @@ int expandsunent(const char *src, char *dst, const char *key,
 						dst += l;
 					}
 					len += l;
-				} else
-					return 0;
+				}
 				src = p;
 			}
 			break;
